### PR TITLE
Added update to referral sources

### PR DESCRIPTION
**Pull request by hagen**

### CODE REVIEW

**Comment by hagen** [comment]
Oh yeh. Editor is set to four, but *does* has auto-detect on. I find, anecdotally, that vs code has trouble in this domain. Also, slightly different flavour of md was used (auto-formatted).

File: sections/referral_sources.md [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+-   [Update Referral Source](#update-referral-source "This update a patient's referral source.")
 
-Get Referral Sources
-------------
+## Get Referral Sources
 
 **Resources**
-* ```GET /referral_sources``` get all referral sources
+
+-   `GET /referral_sources` get all referral sources
```
Comment (JohnColvin) [inline sections/referral_sources.md]: extra whitespace

File: sections/referral_sources.md [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
-============
+# Referral Source
+
 > Referral source of a patient
 
-* [Get Referral Sources](#get-referral-sources "This will return all referral sources.")
-* [Get Referral Source](#get-referral-source "This will return a patient's referral source.")
+-   [Get Referral Sources](#get-referral-sources 'This will return all referral sources.')
+-   [Get Referral Source](#get-referral-source "This will return a patient's referral source.")
+-   [Update Referral Source](#update-referral-source "This update a patient's referral source.")
```
Comment (JohnColvin) [inline sections/referral_sources.md]: extra whitespace

File: sections/referral_sources.md [hunk 3]
```diff
@@ -54,14 +57,16 @@ curl https://api.cliniko.com/v1/referral_sources \
...
 
-
-Get Referral Source
-------------
+## Get Referral Source
 
 **Resources**
-* ```GET /patients/:patient_id/referral_source``` get a patient's referral source
+
+-   `GET /patients/:patient_id/referral_source` get a patient's referral source
```
Comment (JohnColvin) [inline sections/referral_sources.md]: extra whitespace

File: sections/referral_sources.md [hunk 4]
```diff
@@ -170,3 +179,64 @@ There are a few possible types of referral sources. Depending on the type, diffe
   }
 }
 ```
+
+## Update Referral Source
+
+**Resources**
+
+-   `PUT /patients/:patient_id/referral_source` update a patient's referral source
```
Comment (JohnColvin) [inline sections/referral_sources.md]: extra whitespace

**Pull request merged**